### DISCUSSION TITLE
Tag field depends on

### DISCUSSION THREAD

**Discussion by vitaliyb** [laravel/nova-issues#6484] [Ideas & Feature Requests]
The Tag field is great if you want to allow users to select multiple values.
Sometimes, I need to modify the set of data depending on other fields.

For example, I have a `Tag::make('Favourite Food')` Field where the user selects "Ice cream" and "Pancakes". Next, I have a `Tag::make('Food Brands')` Field where I want the user to select their favorite food brands, but only those linked to the production of "Ice cream" and "Pancakes".

I can't achieve this with the Nova Tag Field since it "does not live-report changes to the Tag field" - as written in the [docs](https://nova.laravel.com/docs/resources/fields.html#supported-dependent-fields).

I wonder why it is like that? What is the reason behind the decision not to live-report Tag field changes?
I find it very strange because I believe that if a field's value may change, it should also be able to report about it. Why not?

Of course, I could use the `MultiSelect` field, but let's be frank about it, nobody uses it because its user experience is far from good. Besides, it does not support `searchable()`.

I could also use the `BooleanGroup` field, but then I lose the ability to search by typing the values I want to select, and it displays all values in a single list that may take up the whole screen.

My proposal, please:

- Allow fields to depend on the Tag field.
- Provide any explanation as to why this has not been done yet.

**TL;DR:**
- Fields should be able to depend on the Tag field.
